p8e-common/src/main/kotlin/io/p8e/definition/DefinitionService.kt

```diff
@@ -86,38 +91,44 @@ class DefinitionService(
     fun loadProto(
         keyPair: KeyPair,
         location: Location,
+        signer: SignerImpl,
         signaturePublicKey: PublicKey? = null
```

Comment:
seems like maybe we could avoid passing the signaturePublicKey through all these functions since it can be fetched from the SignerImpl, right?

Reply:
was hesitant to do this, as some method calls don't pass a public key on purpose thats why i passed at the calling level